kuma/wiki/models.py

```diff
@@ -1295,52 +1295,16 @@ def show_toc(self):
     def language(self):
         return get_language_mapping()[self.locale.lower()]
 
-    def get_absolute_url(self):
+    def get_absolute_url(self, endpoint='wiki.document'):
         """
         Build the absolute URL to this document from its full path
         """
-        return reverse('wiki.document', locale=self.locale, args=[self.slug])
-
-    @staticmethod
-    def from_url(url, required_locale=None, id_only=False):
-        """
-        Return the approved Document the URL represents, None if there isn't
-        one.
+        return reverse(endpoint, locale=self.locale, args=[self.slug])
 
-        Return None if the URL is a 404, the URL doesn't point to the right
-        view, or the indicated document doesn't exist.
+    def get_edit_url(self):
```

Comment:
One day I want to centralize the URL construction for the domain objects so we can do `document.urls.edit()` or something, since we do *a lot* of URL reversal during runtime.